vault/resource_kv_secret_v2.go

```diff
@@ -184,13 +261,72 @@ func kvSecretV2Read(_ context.Context, d *schema.ResourceData, meta interface{})
 				if err := d.Set(consts.FieldMetadata, serializeDataMapToString(v)); err != nil {
 					return diag.FromErr(err)
 				}
+
+				// Read & Set custom metadata
+				if _, ok := v[consts.FieldCustomMetadata]; ok {
+					cm, err := readKVV2Metadata(d, client)
+					if err != nil {
+						return diag.FromErr(err)
+					}
+
+					customMetadata := []interface{}{cm}
+					if err := d.Set(consts.FieldCustomMetadata, customMetadata); err != nil {
+						return diag.FromErr(err)
+					}
+				}
 			}
 		}
+
 	}
 
 	return nil
 }
 
+func readKVV2Metadata(d *schema.ResourceData, client *api.Client) (map[string]interface{}, error) {
+	mount := d.Get(consts.FieldMount).(string)
+	name := d.Get(consts.FieldName).(string)
+
+	path := getKVV2MetadataPath(mount, name)
+
+	log.Printf("[DEBUG] Reading metadata for KVV2 secret at %s", path)
+	resp, err := client.Logical().Read(path)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp == nil {
+		log.Printf("[DEBUG] no metadata found for secret")
+		return nil, nil
+	}
+
+	metadataFields := map[string]string{
```

Comment:
This map may be useful as a global that can be used to replace the duplicate remapping logic in `getCustomMetadata()`, and any where else you are accessing these fields.

Reply:
Ah good call, thanks for the suggestion! Updated the code and removed the duplication